common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -473,6 +474,16 @@ def _fetch_file_content_from_github(path, repo, commit)
       end
       # rubocop:enable Metrics/AbcSize
 
+      def cloned_commit
+        return if repo_contents_path.nil? || !File.directory?(File.join(repo_contents_path, ".git"))
```

Comment:
I noticed that this logic is now duplicated with `Job#already_cloned?`. That makes me wonder (other than whether that shows any refactoring opportunities), does it make sense to also add this extra check to other places in the fetcher that check for `repo_contents_path.nil?`?

Reply:
Yeah it was an awkward refactor too since we need to check before constructing a file_fetcher, and the file_fetcher doesn't get the job so I can't just call the method on the job. I punted and opted for the duplication for now.

I think since this is just an error path any other checks for repo_contents_path is safe? If we get past the clone step without throwing we can safely assume the repo exists there now.